src/pybind/mgr/dashboard_v2/restresource.py

```diff
@@ -8,12 +10,23 @@ def inner(*args, **kwargs):
         content_length = int(cherrypy.request.headers['Content-Length'])
         body = cherrypy.request.body.read(content_length)
         if not body:
-            raise cherrypy.HTTPError(400, 'Empty body. Content-Length={}'.format(content_length))
+            raise cherrypy.HTTPError(400, 'Empty body. Content-Length={}'
+                                          .format(content_length))
         try:
             data = json.loads(body.decode('utf-8'))
+            f_args = inspect.getargspec(func).args
+            n_args = []
+            for arg in args:
+                n_args.append(arg)
+            for arg in f_args[1:]:
+                if arg in data:
+                    n_args.append(data[arg])
+                    data.pop(arg)
+            kwargs.update(data)
```

Comment:
This is imo problematic:

1. For our NoDB models, I will have to reverse this. Converting the data to function arguments here is the wrong place for me.
1. `inspect.getargspec(func).args` is unreliable. Especially in combination with decorators.

Reply:
Can we leave it as it is now and change in the future to cope with nodb stuff?